aretomo/protocols/protocol_aretomo.py

```diff
@@ -522,25 +612,66 @@ def createOutputStep(self, tsObjId: int, tsId: str, tsFn: str):
             outputSetOfTiltSeries.write()
             self._store(outputSetOfTiltSeries)
 
+            # Output set of CTF tomo series
+            if Plugin.getActiveVersion() != V1_3_4 and self.doEstimateCtf:
+                outputCtfs = self.getOutputSetOfCtfs()
+
+                newCTFTomoSeries = CTFTomoSeries()
+                newCTFTomoSeries.copyInfo(newTs)
+                newCTFTomoSeries.setTiltSeries(newTs)
+                #newCTFTomoSeries.setObjId(ts.getObjId())
+                newCTFTomoSeries.setTsId(tsId)
+                outputCtfs.append(newCTFTomoSeries)
+
+                outputFile = self.getFilePath(tsFn, extraPrefix, "_ctf.txt")
+                ap = AretomoCtfParser(self)
+                ap.parseTSDefocusFile(newTs, outputFile, newCTFTomoSeries)
+
+                outputCtfs.update(newCTFTomoSeries)
+                outputCtfs.write()
+                self._store(outputCtfs)
+
+    def _closeOutputSet(self):
+        super()._closeOutputSet()
+        if self.makeTomo:
+            outSet = getattr(self, OUT_TOMO, [])
+        else:
+            outSet = getattr(self, OUT_TS, [])
+        if len(outSet) == 0:
+            msg = ('All the introduced Tilt Series were not correctly aligned and/or all the corresponding '
+                   'tomograms were not correctly generated.')
+            self.noneGeneratedMsg.set(msg)
+            raise Exception(msg)
+
     # --------------------------- INFO functions ------------------------------
     def _summary(self) -> List[str]:
         summary = []
         if hasattr(self, OUT_TOMO):
             summary.append(f"Input tilt-series: "
                            f"{self._getSetOfTiltSeries().getSize()}.\n"
                            f"Tomograms reconstructed: "
-                           f"{self.outputSetOfTomograms.getSize()}.\n")
+                           f"{getattr(self, OUT_TOMO).getSize()}.\n")
         elif hasattr(self, OUT_TS):
             summary.append(f"Input tilt-series: "
                            f"{self._getSetOfTiltSeries().getSize()}.\n"
                            f"Tilt series aligned: "
-                           f"{self.outputSetOfTiltSeries.getSize()}")
+                           f"{getattr(self, OUT_TS).getSize()}")
         else:
             summary.append("Output is not ready yet.")
 
-        if self._saveInterpolated() and not self.makeTomo:
-            summary.append("*Interpolated TS stack may have a few "
-                           "dark tilt images removed.*")
+        if self.isFinished() and self.noneGeneratedMsg.get():
+            summary.append('*ERROR!*\n' + self.noneGeneratedMsg.get())
```

Comment:
I'm still getting this error msg printed even when all outputs are produced correctly.

Reply:
Hi, @azazellochg. I'm not able to reproduce it. Could it be related to a missing git pull or something like that?

Reply:
Okay, let me try again this evening

Reply:
Indeed I'm not getting it. Now I realize that this code will never be executed.
```
self.noneGeneratedMsg.set(msg)
raise Exception(msg)
```
will raise an exception and fail the protocol, so the status will never be finished

Reply:
Thanks, @azazellochg. The Exception is the desired behaviour as no protocol should finish its execution in green with a correct status if no output is generated (when there are outputs expected to be generated, of course). You are right regarding the summary message, it's dead code from a previous attempt. Let me delete those lines.

Reply:
On the other hand, are you getting the message no matter how the execution goes, as said previously? I'm not getting it. This is what I get from the tests.

![image](https://github.com/scipion-em/scipion-em-aretomo/assets/54907551/449bdc3e-157a-49e8-a4c2-d949b0cc2b80)

Or this if some fail, but not all:

![image](https://github.com/scipion-em/scipion-em-aretomo/assets/54907551/bdfb304a-c033-4dc2-af63-0e65b410615a)

Reply:
Nope, it's all correct. I get the same as you now.